retrieval/series_cache.go

```diff
@@ -356,11 +367,20 @@ func (c *seriesCache) lookup(ctx context.Context, ref uint64) (retErr error) {
 	c.mtx.Unlock()
 
 	if entry.lset == nil {
-		// in which case the entry did not match the filters
+		// in which case the entry did not match the filters.  it was
+		// added to failingReporter in set().
 		return nil
 	}
 
-	defer seriesCacheLookupCounter.Add(ctx, 1, &retErr)
+	failedReason := "unknown"
```

Comment:
Not a huge deal either ways, but i wonder if the reasons for failures should be defined together somewhere, i'm assuming i'll be grepping through the code soon enough for them 😄

Reply:
Even if we document the local conditions precisely, (a) I was hoping they'd be self-explanatory, (b) there are a bunch of server-side conditions that won't be documented here anyway.